psycop/common/cohort_definition.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+import polars as pl
+from pydantic import BaseModel
+
+
+@dataclass
+class StepDelta:
+    step_name: str
+    n_before: int
+    n_after: int
+
+    @property
+    def n_dropped(self) -> int:
+        return self.n_before - self.n_after
+
+
+class FilteredCohort(BaseModel):
+    """A filtered cohort is a cohort that has been filtered by a set of criteria."""
+
+    cohort: pl.DataFrame
```

Comment:
Might want to represent the granularity here (e.g. this is not at the level of patients). For example:
```suggestion
    prediction_times: pl.DataFrame
```